guides/Scripting_Languages.md

```diff
@@ -0,0 +1,320 @@
+Chapters are sorted in alphabetical order.
+A dialect means the language is inspired by the parent but not fully syntax compatible.
+
+Fennel (Lisp/Clojure dialect)
+---
+
+<img align="right" width="300" height="300" src="https://fennel-lang.org/logo.svg" alt="Fennel Logo">
+
+* Home: [Fennel](https://fennel-lang.org/)
+* Source: [github](https://github.com/bakpakin/Fennel)
+* Syntax: [Lisp](https://en.wikipedia.org/wiki/Lisp_\(programming_language\))/[Clojure](https://clojure.org/) inspired
+* Semantic: mostly Lua
+* jit compiler: Pure Lua
+* Dependencies: (TODO)
+
+#### ahead-of-time
+
+    $ fennel --compile yourProject
+
+(TODO: error line translation)
+
+#### just-in-time
+
+* [Setup#Embedding Fennel](https://fennel-lang.org/setup#embedding-fennel)
+* [Minimal LÖVR Setup](https://github.com/MattRoelle/min-lovr-fennel)
+
+Haxe (ECMAScript dialect)
+---
+
+<img align="right" width="200" height="200" src="https://haxe.org/img/haxe-logo.svg" alt="Haxe Logo">
+
+* Home: [Haxe](https://haxe.org/)
+* Source: [github](https://github.com/HaxeFoundation/haxe)
+* Syntax: [ECMAScript](https://en.wikipedia.org/wiki/ECMAScript) inspired
+* Semantic: (TODO)
+* jit compiler: Not available (TODO: is that true
+* Dependencies: (TODO)
+
+#### ahead-of-time
+
+[Haxe Lua](https://haxe.org/manual/target-lua.html)
+
+(TODO: Line Translation)
+
+#### just-in-time
+
+Haxe does not offer runtime compilation. (TODO: is that true?)
+
+Moonscript (CoffeeScript dialect) 
+---
+
+<img align="right" width="300" height="150" src="http://moonscript.org/images/logo.png" alt="Moonscript Logo">
+
+* Home: [Moonscript](https://www.moonscript.org)
+* Source: [github](https://github.com/leafo/moonscript)
+* Syntax: [CoffeeScript](https://coffeescript.org) inspired [(white space sensitive)](https://en.wikipedia.org/wiki/Off-side_rule)
+* Semantic: Lua
+* jit compiler: Lua + Lua C Library (LPeg) 
+* Dependencies: [LPeg](http://www.inf.puc-rio.br/~roberto/lpeg/)
+* Features:
+    * Mature
+    * Class system for object oriented coding
+
+#### ahead-of-time
+
+The standalone compiler can be used to 
+
+    $ moonc yourGame
+
+(TODO: Write more about error line rewriting.)
+
+
+#### just-in-time
+
+Setup in conf.lua:
+
+``` lua
+-- This allows us to have a dedicated directory for Lua c libraries.
+local os = lovr.system.getOS()
+if os == 'Linux' then
+    package.cpath = 'utils/lua_clib/?.so'
+end
+if os == 'Windows' then
+    package.cpath = 'utils/lua_clib/?.dll'
+end
+if os == 'macOS' then
+    package.cpath = 'utils/lua_clib/?.so'
+end
+
+-- Adjust the require search paths to your needs.
+local lua_path = './?.lua;./?/init.lua;utils/moonscript/?/init.lua;'
+lovr.filesystem.setRequirePath(lua_path)
+
+-- Moonscript must be located at utils. Check for utils/moonscript/moonscript/init.lua.
+-- The LPeg library is to be located at:
+-- Linux:    utils/lua_clib/lpeg.so
+-- Windows:  utils\lua_clib\lpeg.dll
+-- MacOS:    utils/lua_clib/lpeg.so
```

Comment:
LÖVR is compiled so that it can always load shared libraries next to the executable, and it adds a `require` loader that will search for libraries next to the executable.  This is the basis of LÖVR's plugin system.

Practically, it means that for C libraries like LPeg/yue you can place it next to the lovr executable and don't need to modify `package.cpath`.  Also `package.cpath` depends on the working directory matching lovr.filesystem root.

This may make some of the instructions in the guide a bit simpler.

Reply:
I know about ḶÖVR's default behavior but thought it is a good idea to teach the reader that he can put the binaries in a dedicated folder.

Albeit your mentioning of working directory == lovr.filesystem root makes me think otherwise.
What do you think?

Reply:
I guess I am going to remove the mangling with package.cpath and just write a general section explaining Lua C libs.

Reply:
There is already a page that describes how to use C libraries with LÖVR (Plugins)